tests/testthat/test-fbref.R

```diff
@@ -340,3 +340,179 @@ test_that("fb_player_goal_logs() works", {
   expect_false(nrow(jwp_goal_log) == 0)
   expect_false(nrow(gnahoua_goal_log) == 0)
 })
+
+Sys.sleep(3)
+test_that("fb_league_stats() works", {
+  testthat::skip_on_cran()
+
+  expected_shooting_cols <- c("Team_or_Opponent", "Squad", "Num_Players", "Mins_Per_90", "Gls_Standard", "Sh_Standard", "SoT_Standard", "SoT_percent_Standard", "Sh_per_90_Standard", "SoT_per_90_Standard", "G_per_Sh_Standard", "G_per_SoT_Standard", "Dist_Standard", "FK_Standard", "PK_Standard", "PKatt_Standard", "xG_Expected", "npxG_Expected", "npxG_per_Sh_Expected", "G_minus_xG_Expected", "np:G_minus_xG_Expected", "url")
+  epl_team_shooting_22 <- fb_league_stats(
+    country = "ENG",
+    gender = "M",
+    season_end_year = 2022,
+    tier = "1st",
+    team_or_player = "team",
+    stat_type = "shooting"
+  )
+  expect_gt(nrow(epl_team_shooting_22), 0)
+  expect_equal(sort(colnames(epl_team_shooting_22)), sort(expected_shooting_cols))
+
+  expected_misc_cols <- c("Team_or_Opponent", "Squad", "Num_Players", "Mins_Per_90", "CrdY", "CrdR", "2CrdY", "Fls", "Fld", "Off", "Crs", "Int", "TklW", "PKwon", "PKcon", "OG", "Recov", "Won_Aerial Duels", "Lost_Aerial Duels", "Won_percent_Aerial Duels", "url")
+  ## testing a lot would take too long, so just test multiple stat years since that is the most likely input param to have multiple values
+  multi_league_season_tier_misc <- fb_league_stats(
+    # country = c("ITA", "ESP"),
+    country = "ESP",
+    # gender = c("M", "F"),
+    gender = "M",
+    season_end_year = 2021:2022,
+    # tier = c("1st", "2nd"),
+    tier = "1st",
+    team_or_player = "team",
+    stat_type = "misc"
+  )
+  expect_gt(nrow(multi_league_season_tier_misc), 0)
+  expect_equal(sort(colnames(multi_league_season_tier_misc)), sort(expected_misc_cols))
+  expect_equal(length(unique(multi_league_season_tier_misc$url)), 2)
+
+  expect_error(
+    fb_league_stats(
+      country = "England",
+      gender = "M",
+      season_end_year = 2022,
+      tier = "1st",
+      team_or_player = "team",
+      stat_type = "shooting"
+    ),
+    regexp = "input"
+  )
+
+  expect_error(
+    fb_league_stats(
+      country = "ENG",
+      gender = NA_character_,
+      season_end_year = 2022,
+      tier = "1st",
+      team_or_player = "team",
+      stat_type = "shooting"
+    ),
+    regexp = "input"
+  )
+
+  expect_error(
+    fb_league_stats(
+      country = "ENG",
+      season_end_year = 2022,
+      tier = "1st",
+      team_or_player = "team",
+      stat_type = "shooting"
+    ),
+    regexp = "gender.*missing"
+  )
+
+  expect_error(
+    fb_league_stats(
+      country = "ENG",
+      gender = "M",
+      season_end_year = 2032,
+      tier = "1st",
+      team_or_player = "team",
+      stat_type = "shooting"
+    ),
+    regexp = "input"
+  )
+
+  expect_error(
+    fb_league_stats(
+      country = "ENG",
+      gender = "M",
+      tier = "1st",
+      team_or_player = "team",
+      stat_type = "shooting"
+    ),
+    regexp = "season_end_year"
+  )
```

Comment:
i don't think we thoroughly test `country`, `gender`, `tier` and `season_end_year` elsewhere, so i went ahead and did it here. in the future we might consider validation in functions that take in these parameters, such as

```r

purrr::map_chr(
  country,
  ~rlang::arg_match0(
    .x,
    unique(seasons$country)
  )
)

purrr::map_chr(
  gender,
  ~rlang::arg_match0(
    .x,
    unique(seasons$country)
  )
)

purrr::map_chr(
  as.character(season_end_year),
  ~rlang::arg_match0(
    .x,
    as.character(unique(seasons$season_end_year))
  )
)


purrr::map_chr(
  tier,
  ~rlang::arg_match0(
    .x,
    unique(seasons$tier)
  )
)

if (!is.na(non_dom_league_url)) {
  purrr::map_chr(
    non_dom_league_url,
    ~rlang::arg_match0(
      .x,
      unique(seasons$comp_url)
    )
  )
}
```

Reply:
Nice